### PR TITLE
moved error code enumerations to the result of action messages

### PR DESCRIPTION
**Pull request by jwallace42**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | None |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | Gazebo |

---

## Description of contribution in a few bullet points
Previously, the error code enumerations were define in the goal section of the action message. This was incorrect because the the error code is a result. This pr moves the error code enumerations for the actions to the result section.  

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_msgs/action/SmoothPath.action [hunk 1]
```diff
@@ -1,5 +1,12 @@
-# Error codes
-# Note: The expected priority order of the errors should match the message order
+#goal definition
```
Comment (SteveMacenski) [inline nav2_msgs/action/SmoothPath.action:1]: All of the action definitions need proper capitalization + a space between the `#` and the comment for the changes. I'm not sure how those got that way with just copy pasting :wink:

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
I generally approve, seems like a reasonable change!

File: nav2_msgs/action/SmoothPath.action (hunk 1 above)
Comment (jwallace42) [reply nav2_msgs/action/SmoothPath.action:1]: I added a space and capitalized the `error codes` and `note`. The definitions were always lower case. 

i.e `#goal definition`

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
@jwallace42 can you open a PR for migration guide?

**Pull request merged**